subsys/bluetooth/audio/vcs_client.c

```diff
@@ -724,7 +724,11 @@ int bt_vcs_discover(struct bt_conn *conn)
 
 	if (conn == NULL) {
```

Comment:
Unrelated to this PR, but a couple of questions come to mind regarding the above line:

 - Shouldn't it be using `CHECKIF()`?
 - Why is it returning `-ENOTCONN` instead of `-EINVAL`? I'd expect `-ENOTCONN` to be returned only if `conn->state == BT_CONN_DISCONNECTED`

Reply:
And to follow up: if disconnected state is indeed something to check for isn't there a missing check for `conn->state` in that case?

Reply:
1) Yes it should have been CHECKIF
2) Yes, it should return -EINVAL instead of -ENOTCONN (needs to be fixed many other places, but will do that in a separate PR)
3) I don't think we need to check for connected state at this level, as `bt_gatt_discover` will do that for us.